airflow-3.yaml

```diff
@@ -0,0 +1,267 @@
+      pip${{vars.py-version}} install --verbose \
+       --prefix="/opt/airflow" --root=${{targets.contextdir}} dist/*.whl
+
+  - name: "Build and install the additional provider packages"
+    runs: |
+      # By default, the airflow celery provider is not built, but running the upstream helm chart requires it
+      # There are a few others we want to include by default as well
+
+      cd providers
+      # If changing this list, be sure to update the test below too
+      # Use / as a separator
+      for pkg in amazon celery cncf/kubernetes docker elasticsearch google grpc hashicorp microsoft/azure mysql odbc openlineage postgres redis sendgrid sftp slack snowflake ssh; do
+        cd "$pkg"
+        python${{vars.py-version}} -m build --wheel
+        pip${{vars.py-version}} install --verbose \
+          --prefix="/opt/airflow" --root=${{targets.contextdir}} dist/*.whl
+        cd -
+      done
+
+  - runs: find . -name '__pycache__' -exec rm -rf {} +
+
+  - runs: |
+      mkdir -p ${{targets.destdir}}/opt/airflow/dags
+      mkdir -p ${{targets.destdir}}/opt/airflow/logs
+      mkdir -p ${{targets.destdir}}/scripts/docker
+
+      # The first time you run Airflow, it will create a file called `airflow.cfg` in
+      # `$AIRFLOW_HOME` directory
+      # However, for production case it is advised to generate the configuration
+      export PYTHONPATH=${{targets.destdir}}/opt/airflow/lib/python${{vars.py-version}}/site-packages
+
+      ${{targets.destdir}}/opt/airflow/bin/airflow config list --defaults > ${{targets.destdir}}/"airflow.cfg"
+
+      cp airflow-core/src/airflow/config_templates/default_webserver_config.py ${{targets.contextdir}}/
+
+      install -Dm755 scripts/docker/entrypoint_prod.sh ${{targets.destdir}}/entrypoint
+      install -Dm755 scripts/docker/clean-logs.sh ${{targets.destdir}}/clean-logs
+      install -Dm755 scripts/docker/airflow-scheduler-autorestart.sh ${{targets.destdir}}/airflow-scheduler-autorestart
+      install -Dm755 scripts/docker/* ${{targets.destdir}}/scripts/docker
+
+  - if: ${{build.arch}} == "aarch64"
+    name: "Install openssl 1.1 on arm64 for apache-airflow-providers-mysql"
+    runs: |
+      # apache-airflow-providers-mysql links against libcrypto.so.1.1 on arm64, and does not provide the
+      # necessary source files to rebuild the package for arm against libcrypto3 (like the x86 version is),
+      # so instead we'll just quickly build openssl 1.1 for it (since we don't have a wolfi package).
+      wget https://www.openssl.org/source/old/1.1.1/openssl-1.1.1u.tar.gz
+      tar xzf openssl-1.1.1u.tar.gz
+      cd openssl-1.1.1u
+      ./config --prefix=${{targets.destdir}}/usr shared
+      make -j$(nproc)
+      make install_sw # Install only libraries & headers (skip docs)
+
+      # Remove extra links so things still default to the more modern version
+      rm ${{targets.destdir}}/usr/lib/libcrypto.so
+      rm ${{targets.destdir}}/usr/lib/libssl.so
```

Comment:
OpenSSL 1.1.1 isn't receiving security support any longer, so I don't think we can do this. We either need to figure out a way to get it using a modern OpenSSL, or we'll have to drop support for this provider on ARM.

The MySQL database driver situation in Python is confusing (there are a bunch of different ones), but we can dig into it together.